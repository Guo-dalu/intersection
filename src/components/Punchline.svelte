<span class="gradientcolor">
	<slot />
</span>

<style>
	.gradientcolor {
		color: transparent;
		background-clip: text;
		@apply from-pink-600 to-orange-500;
		@apply bg-gradient-to-r;
		@apply font-bold;
	}
</style>
