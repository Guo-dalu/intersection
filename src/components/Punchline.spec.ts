import { expect, describe, it } from 'vitest'
import { render } from '@testing-library/svelte'
import Punchline from './Punchline.svelte'

describe('Punchline Component', () => {
	it('should render correctly', () => {
		// still can not test slot in a normal way, see: https://github.com/testing-library/svelte-testing-library/issues/48
		const { container } = render(Punchline)

		expect(container).toMatchSnapshot()
	})
})
