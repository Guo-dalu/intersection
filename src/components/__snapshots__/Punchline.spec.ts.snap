// Vitest Snapshot v1

exports[`Punchline Component > should render correctly 1`] = `
<body>
  <div>
    <span
      class="gradientcolor s-qaK2xGRD7wDi"
    />
    <!--&lt;Punchline&gt;-->
  </div>
</body>
`;
